src/main/java/io/axoniq/axonserver/connector/event/impl/DcbEventChannelImpl.java

```diff
@@ -0,0 +1,225 @@
+    private final DcbEventStoreGrpc.DcbEventStoreStub eventStore;
+    private final ClientIdentification clientIdentification;
+
+    /**
+     * Instantiate an {@link AbstractAxonServerChannel}.
+     *
+     * @param clientIdentification     the identification of the client
+     * @param executor                 a {@link ScheduledExecutorService} used to schedule reconnections
+     * @param axonServerManagedChannel the {@link AxonServerManagedChannel} used to connect to AxonServer
+     */
+    public DcbEventChannelImpl(ClientIdentification clientIdentification,
+                               ScheduledExecutorService executor,
+                               AxonServerManagedChannel axonServerManagedChannel) {
+        super(clientIdentification, executor, axonServerManagedChannel);
+        this.eventStore = DcbEventStoreGrpc.newStub(axonServerManagedChannel);
+        this.clientIdentification = clientIdentification;
+    }
+
+    @Override
+    public void connect() {
+        // nothing to do here
+    }
+
+    @Override
+    public void reconnect() {
+        // nothing to do here
+    }
+
+    @Override
+    public void disconnect() {
+        // nothing to do here
+    }
+
+    @Override
+    public boolean isReady() {
+        return true;
+    }
+
+    @Override
+    public AppendEventsTransaction startTransaction() {
+        FutureStreamObserver<AppendEventsResponse> response = new FutureStreamObserver<>(null);
+        StreamObserver<AppendEventsRequest> clientStream = eventStore.append(response);
+        return new AppendEventsTransactionImpl(clientStream, response);
+    }
+
+    @Override
+    public ResultStream<StreamEventsResponse> stream(StreamEventsRequest request) {
+        AbstractBufferedStream<StreamEventsResponse, Empty> result =
+                new AbstractBufferedStream<StreamEventsResponse, Empty>(clientIdentification.getClientId(),
+                                                                        BUFFER_SIZE,
+                                                                        REFILL_BATCH) {
+                    @Override
+                    protected Empty buildFlowControlMessage(FlowControl flowControl) {
+                        return null;
+                    }
+
+                    @Override
+                    protected StreamEventsResponse terminalMessage() {
+                        return StreamEventsResponse.newBuilder()
+                                                   .build();
+                    }
+                };
+        eventStore.stream(request, result);
+        return result;
+    }
+
+    @Override
+    public ResultStream<SourceEventsResponse> source(SourceEventsRequest request) {
+        AbstractBufferedStream<SourceEventsResponse, Empty> result =
+                new AbstractBufferedStream<SourceEventsResponse, Empty>(clientIdentification.getClientId(),
+                                                                        BUFFER_SIZE,
```

Comment:
I spotted that the default buffer size for an aggregate event stream is `512` even! Any clue why that's the case? I am wondering whether that discrepancy between the old and new event channel will come to bite us is all.

Reply:
It depends on how fast the client consumes events. The bigger the buffer, the more heap is used.

Reply:
Yes, but you are right, we should use the default that `EventChannel` uses.